### DISCUSSION TITLE
Paper: Let Me Speak Freely? A Study on the Impact of Format Restrictions on Performance of Large Language Models

### DISCUSSION THREAD

**Discussion by ChristianWeyer** [outlines-dev/outlines#1117] [General]
https://arxiv.org/pdf/2408.02442

> Structured generation, the process of producing content in standardized formats like JSON and XML, is widely utilized in real-world applications to extract key output information from large language models (LLMs). This study investigates whether such constraints on generation space impact LLMs’ abilities, including reasoning and domain knowledge comprehension. Specifically, we evaluate LLMs’ performance when restricted to adhere to structured formats versus generating free-form responses across various common tasks. Surprisingly, we observe a significant decline in LLMs’ reasoning abilities under format restrictions. Furthermore, we find that stricter format constraints generally lead to greater performance degradation in reasoning tasks.

Would really love to hear your opinion on this @rlouf et. al..

**Comment by just-cameron** [comment]
I don't particularly like the methodology used here. The structured text they use here is [JSON mode](https://platform.openai.com/docs/guides/structured-outputs/json-mode), not formal structured output. Additionally, the prompts they use don't really advantage structured text at all.

![image](https://github.com/user-attachments/assets/66a695ff-df48-45ea-9c22-b982a95596a6)

Note that they don't really provide a schema here that would help structured text provide a good answer. I think the [CoT example](https://outlines-dev.github.io/outlines/cookbook/chain_of_thought/) in the cookbook is probably a better approach.
